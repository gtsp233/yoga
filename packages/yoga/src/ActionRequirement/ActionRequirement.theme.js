const ActionRequirement = () => ({});

export default ActionRequirement;
