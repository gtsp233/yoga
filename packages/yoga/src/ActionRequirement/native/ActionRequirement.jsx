import React from 'react';
import { arrayOf, node, oneOfType, string } from 'prop-types';
import styled from 'styled-components/native';
import { View } from 'react-native';
import Box from '../../Box';
import { Title } from './ActionRequirementStyles';
import Text from '../../Text';

const StyledActionRequirement = styled.View``;

function ActionRequirement(props) {
  const { title, description, children, checkable, illustration, list } = props;

  return (
    <StyledActionRequirement {...props}>
      {illustration && <Box>{illustration}</Box>}
      <Title>{title}</Title>
      <Text mt="small" color="deep">
        {description}
      </Text>
      <Text mt="small" color="deep">
        {list && <Box mt="xxxlarge">{list}</Box>}
      </Text>
      {checkable && <Box mt="xxxlarge">{checkable}</Box>}
      <View>{children}</View>
    </StyledActionRequirement>
  );
}

ActionRequirement.propTypes = {
  title: string.isRequired,
  children: oneOfType([arrayOf(node), node]),
  description: string.isRequired,
  checkable: oneOfType([arrayOf(node), node]),
  illustration: oneOfType([arrayOf(node), node]),
  list: oneOfType([arrayOf(node), node]),
};

ActionRequirement.defaultProps = {
  children: undefined,
  checkable: undefined,
  illustration: undefined,
  list: undefined,
};

export default ActionRequirement;
