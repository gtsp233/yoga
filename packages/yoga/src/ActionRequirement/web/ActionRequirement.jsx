import React from 'react';
import { arrayOf, node, oneOfType, string } from 'prop-types';
import styled from 'styled-components';
import {
  Actions,
  Title,
  PrimaryButton,
  SecondaryButton,
} from './ActionRequirementStyles';
import Text from '../../Text';
import Box from '../../Box';

const StyledActionRequirement = styled.div``;

function isChildFromComponent(child, component) {
  return child.type.displayName === component.displayName;
}

function ActionRequirement(props) {
  const { title, description, children, checkable, illustration, list } = props;

  let primaryButton;
  let secondaryButton;

  function defineCompoundComponents() {
    React.Children.forEach(children, child => {
      if (isChildFromComponent(child, PrimaryButton)) primaryButton = child;
      if (isChildFromComponent(child, SecondaryButton)) secondaryButton = child;
    });
  }

  defineCompoundComponents();
  return (
    <StyledActionRequirement {...props}>
      {illustration && <Box>{illustration}</Box>}
      <Title>{title}</Title>
      <Text mt="small" color="deep">
        {description}
      </Text>
      <Text mt="small" color="deep">
        {list && <Box mt="xxxlarge">{list}</Box>}
      </Text>
      {checkable && <Box mt="xxxlarge">{checkable}</Box>}
      <Actions mt="xlarge">
        {primaryButton}
        {secondaryButton}
      </Actions>
    </StyledActionRequirement>
  );
}

ActionRequirement.propTypes = {
  title: string.isRequired,
  children: oneOfType([arrayOf(node), node]),
  description: string.isRequired,
  checkable: oneOfType([arrayOf(node), node]),
  illustration: oneOfType([arrayOf(node), node]),
  list: oneOfType([arrayOf(node), node]),
};

ActionRequirement.defaultProps = {
  children: undefined,
  checkable: undefined,
  illustration: undefined,
  list: undefined,
};

export default ActionRequirement;
