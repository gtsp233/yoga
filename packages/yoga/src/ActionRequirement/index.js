import ActionRequirement from './web';

export default ActionRequirement;
