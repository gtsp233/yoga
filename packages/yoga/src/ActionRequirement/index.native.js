import ActionRequirement from './native';

export default ActionRequirement;
