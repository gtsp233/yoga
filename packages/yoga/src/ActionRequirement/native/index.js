import ActionRequirement from './ActionRequirement';
import {
  Title,
  PrimaryButton,
  SecondaryButton,
} from './ActionRequirementStyles';

ActionRequirement.Title = Title;
ActionRequirement.PrimaryButton = PrimaryButton;
ActionRequirement.SecondaryButton = SecondaryButton;

export default ActionRequirement;
