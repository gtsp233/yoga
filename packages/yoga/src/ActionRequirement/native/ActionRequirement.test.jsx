import React from 'react';
import { render } from '@testing-library/react-native';
import { Img2d01 } from '@gympass/yoga-illustrations';
import { Dumbbell, User, Star } from '@gympass/yoga-icons';
import { Box, Checkbox, Text, ThemeProvider } from '../..';

import ActionRequirement from './ActionRequirement';

function handleGetCheckableContent() {
  return (
    <Box display="flex" flexDirection="row" alignItems="center">
      <Checkbox />
      <Text>I have read and agree to the Terms of Use and Privacy Policy.</Text>
    </Box>
  );
}

function handleGetIllustration() {
  return <Img2d01 width={375} height={375} />;
}

function handleGetList() {
  return (
    <>
      <Text>
        <Dumbbell /> 1x/day standard access
      </Text>
      <Text mt="small">
        <User /> 4x/month 1-on-1 sessions
      </Text>
      <Text mt="small">
        <Star /> 2x/month premium classes
      </Text>
    </>
  );
}

describe('<ActionRequirement />', () => {
  it('should match snapshot', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <ActionRequirement
          title="Welcome to the world of feeling good!"
          description="Lets make it a good day!"
          illustration={handleGetIllustration()}
          checkable={handleGetCheckableContent()}
          list={handleGetList()}
        />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
