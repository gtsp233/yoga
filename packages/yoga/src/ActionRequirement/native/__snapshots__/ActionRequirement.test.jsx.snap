// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionRequirement /> should match snapshot 1`] = `
<View
  checkable={
    <Styled(View)
      alignItems="center"
      display="flex"
      flexDirection="row"
    >
      <WithTheme(Checkbox) />
      <Text
        inverted={false}
        light={false}
        size="medium"
      >
        I have read and agree to the Terms of Use and Privacy Policy.
      </Text>
    </Styled(View)>
  }
  description="Lets make it a good day!"
  illustration={
    <Unknown
      height={375}
      width={375}
    />
  }
  list={
    <React.Fragment>
      <Text
        inverted={false}
        light={false}
        size="medium"
      >
        <Unknown />
         1x/day standard access
      </Text>
      <Text
        inverted={false}
        light={false}
        mt="small"
        size="medium"
      >
        <Unknown />
         4x/month 1-on-1 sessions
      </Text>
      <Text
        inverted={false}
        light={false}
        mt="small"
        size="medium"
      >
        <Unknown />
         2x/month premium classes
      </Text>
    </React.Fragment>
  }
  style={
    Array [
      Object {},
    ]
  }
  title="Welcome to the world of feeling good!"
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <RNSVGSvgView
      bbHeight={375}
      bbWidth={375}
      focusable={false}
      height={375}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          undefined,
          Object {
            "opacity": 1,
          },
          Object {
            "flex": 0,
            "height": 375,
            "width": 375,
          },
        ]
      }
      width={375}
    >
      <RNSVGGroup
        fill={
          Array [
            0,
            4278190080,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={Array []}
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      />
    </RNSVGSvgView>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 24,
          "fontWeight": "500",
          "lineHeight": 32,
        },
      ]
    }
  >
    Welcome to the world of feeling good!
  </Text>
  <Text
    color="deep"
    inverted={false}
    light={false}
    mt="small"
    size="medium"
    style={
      Array [
        Object {
          "color": "#6B6B78",
          "fontFamily": "Rubik",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
          "marginTop": 16,
        },
      ]
    }
  >
    Lets make it a good day!
  </Text>
  <Text
    color="deep"
    inverted={false}
    light={false}
    mt="small"
    size="medium"
    style={
      Array [
        Object {
          "color": "#6B6B78",
          "fontFamily": "Rubik",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
          "marginTop": 16,
        },
      ]
    }
  >
    <View
      mt="xxxlarge"
      style={
        Array [
          Object {
            "marginTop": 56,
          },
        ]
      }
    >
      <Text
        inverted={false}
        light={false}
        size="medium"
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
          ]
        }
      >
        <RNSVGSvgView
          focusable={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              undefined,
              Object {
                "opacity": 1,
              },
              null,
            ]
          }
        >
          <RNSVGGroup
            fill={
              Array [
                0,
                4278190080,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={Array []}
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          />
        </RNSVGSvgView>
         1x/day standard access
      </Text>
      <Text
        inverted={false}
        light={false}
        mt="small"
        size="medium"
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
              "marginTop": 16,
            },
          ]
        }
      >
        <RNSVGSvgView
          focusable={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              undefined,
              Object {
                "opacity": 1,
              },
              null,
            ]
          }
        >
          <RNSVGGroup
            fill={
              Array [
                0,
                4278190080,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={Array []}
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          />
        </RNSVGSvgView>
         4x/month 1-on-1 sessions
      </Text>
      <Text
        inverted={false}
        light={false}
        mt="small"
        size="medium"
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
              "marginTop": 16,
            },
          ]
        }
      >
        <RNSVGSvgView
          focusable={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              undefined,
              Object {
                "opacity": 1,
              },
              null,
            ]
          }
        >
          <RNSVGGroup
            fill={
              Array [
                0,
                4278190080,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={Array []}
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          />
        </RNSVGSvgView>
         2x/month premium classes
      </Text>
    </View>
  </Text>
  <View
    mt="xxxlarge"
    style={
      Array [
        Object {
          "marginTop": 56,
        },
      ]
    }
  >
    <View
      alignItems="center"
      display="flex"
      flexDirection="row"
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
          },
        ]
      }
    >
      <View>
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              Object {},
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 24,
                  "justifyContent": "center",
                  "width": 24,
                },
              ]
            }
          >
            <View
              checked={false}
              disabled={false}
              inverted={false}
              pressed={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderColor": "#D8385E",
                    "borderRadius": 4,
                    "borderStyle": "solid",
                    "borderWidth": 2,
                    "height": 24,
                    "position": "absolute",
                    "width": 24,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
      <Text
        inverted={false}
        light={false}
        size="medium"
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
          ]
        }
      >
        I have read and agree to the Terms of Use and Privacy Policy.
      </Text>
    </View>
  </View>
  <View />
</View>
`;
